panther_lights/test/test_segment_converter.cpp

```diff
@@ -64,71 +64,122 @@ YAML::Node TestSegmentConverter::CreateSegmentDescription(
 YAML::Node TestSegmentConverter::CreateImageAnimationDescription()
 {
   return YAML::Load(
-    "{type: panther_lights::ImageAnimation, "
-    "image: $(find panther_lights)/animations/triangle01_red.png, "
+    "{image: $(find panther_lights)/animations/triangle01_red.png, "
     "duration: 2}");
 }
 
 TEST_F(TestSegmentConverter, ConvertInvalidChannel)
 {
-  segments_.push_back(
+  segments_.emplace(
+    "name",
     std::make_shared<panther_lights::LEDSegment>(CreateSegmentDescription(0, 10, 123), 50.0));
   const auto anim_desc = CreateImageAnimationDescription();
-  ASSERT_NO_THROW(segments_.at(0)->SetAnimation(anim_desc));
+  ASSERT_NO_THROW(
+    segments_.at("name")->SetAnimation("panther_lights::ImageAnimation", anim_desc, false));
 
   EXPECT_THROW(segment_converter_->Convert(segments_, led_panels_), std::out_of_range);
 }
 
 TEST_F(TestSegmentConverter, ConvertInvalidLedRange)
 {
-  segments_.push_back(std::make_shared<panther_lights::LEDSegment>(
-    CreateSegmentDescription(panel_1_num_led_, panel_1_num_led_ + 10, 1), 50.0));
+  segments_.emplace(
+    "name", std::make_shared<panther_lights::LEDSegment>(
+              CreateSegmentDescription(panel_1_num_led_, panel_1_num_led_ + 10, 1), 50.0));
```

Comment:
I think it's best to leave the smallest value that should causes the error
```suggestion
              CreateSegmentDescription(panel_1_num_led_, panel_1_num_led_ + 1, 1), 50.0));
```